storage2/src/storage.rs

```diff
@@ -0,0 +1,227 @@
+                            db.put_cf(jmt_cf, key_bytes, &value_bytes)?;
+                        }
+
+                        Ok(())
+                    })
+                })
+                .unwrap()
+                .await
+                .unwrap()
+        })
+    }
+}
+
+/// A reader interface for rocksdb. NOTE: it is up to the caller to ensure consistency between the
+/// rocksdb::DB handle and any write batches that may be applied through the writer interface.
+impl TreeReader for Storage {
+    /// Gets node given a node key. Returns `None` if the node does not exist.
+    fn get_node_option<'future, 'a: 'future, 'n: 'future>(
+        &'a self,
+        node_key: &'n NodeKey,
+    ) -> BoxFuture<'future, Result<Option<Node>>> {
+        let db = self.0.clone();
+        let node_key = node_key.clone();
+
+        let span = Span::current();
+
+        Box::pin(async {
+            tokio::task::Builder::new()
+                .name("Storage::get_node_option")
+                .spawn_blocking(move || {
+                    span.in_scope(|| {
+                        let jmt_cf = db.cf_handle("jmt").expect("jmt column family not found");
+                        let value = db
+                            .get_pinned_cf(jmt_cf, &node_key.encode()?)?
+                            .map(|db_slice| Node::decode(&db_slice))
+                            .transpose()?;
+
+                        tracing::trace!(?node_key, ?value);
+                        Ok(value)
+                    })
+                })
+                .unwrap()
+                .await
+                .unwrap()
+        })
+    }
+
+    fn get_rightmost_leaf<'future, 'a: 'future>(
+        &'a self,
+    ) -> BoxFuture<'future, Result<Option<(NodeKey, jmt::storage::LeafNode)>>> {
+        let span = Span::current();
+        let db = self.0.clone();
+
+        Box::pin(async {
+            tokio::task::Builder::new()
+                .name("Storage::get_rightmost_leaf")
+                .spawn_blocking(move || {
+                    span.in_scope(|| {
+                        let jmt_cf = db.cf_handle("jmt").expect("jmt column family not found");
+                        let mut iter = db.raw_iterator_cf(jmt_cf);
+                        let mut ret = None;
+                        iter.seek_to_last();
+
+                        if iter.valid() {
+                            let node_key = NodeKey::decode(iter.key().unwrap())?;
+                            let node = Node::decode(iter.value().unwrap())?;
+
+                            if let Node::Leaf(leaf_node) = node {
+                                ret = Some((node_key, leaf_node));
+                            }
+                        } else {
+                            // There are no keys in the database
+                        }
+                        Ok(ret)
+                    })
+                })
+                .unwrap()
+                .await
+                .unwrap()
+        })
+    }
+}
```

Comment:
We shouldn't implement this trait on the `Storage` type, and in any case the trait will have to change, because we'll be using the non-`async` JMT API, so we should delete this implementation.

Reply:
@hdevalence why shouldn't we implement the `TreeReader` trait on `Storage`? We still have need to read the JMT in `Storage`, i.e. to get the JMT version